packages/aws-cdk-lib/aws-ecs/lib/log-drivers/aws-log-driver.ts

```diff
@@ -82,6 +82,16 @@ export interface AwsLogDriverProps {
    * @default - AwsLogDriverMode.BLOCKING
    */
   readonly mode?: AwsLogDriverMode;
+
+  /**
+   * When AwsLogDriverMode.NON_BLOCKING is configured, this parameter
+   * controls the size of the non-blocking buffer used to temporarily
+   * store messages. This parameter is not valid with 
+   * AwsLogDriverMode.BLOCKING.
```

Comment:
Since this is not valid with blocking mode, should I add some sort of check? How does that work? Can someone help me with how to do that- does it go in the `bind` method and what do I make it do/throw/return on failure?

Reply:
Something like? =>

```
if (this.props.maxBufferSize != "" && this.props.mode != AwsLogDriverMode. NON_BLOCKING) {
    throw new Error('max-buffer-size requires non-blocking mode')
}
```

Reply:
Yes i was about to ask for the same thing. It will be great if we can perform a check in L2.

Maybe
```ts
if (this.props.maxBufferSize != "" && this.props.mode === AwsLogDriverMode.BLOCKING) { throw new Error('max-buffer-size is not valid with blocking mode') }
```
which is more related to "This parameter is not valid with AwsLogDriverMode.BLOCKING.", unless you want to change the comment msg to "this parameter is only valid with non-blocking mode".